domain/modelconfig/service/service.go

```diff
@@ -324,7 +330,9 @@ func (s *Service) updateModelConfigValidator(
 		Validators: []config.Validator{
 			validators.AgentVersionChange(),
 			validators.CharmhubURLChange(),
-			validators.SpaceChecker(&dummySpaceProvider{}),
+			validators.SpaceChecker(&spaceValidator{
```

Comment:
I think we should make updateModelConfigValidator take a `context ` that can be passed to the space validator on construction.

Reply:
I answered in https://github.com/juju/juju/pull/17330#issuecomment-2092969819